frontend/src/metabase/dashboard/components/DashCard/DashCardMenu/DashCardMenu.tsx

```diff
@@ -174,7 +174,7 @@ DashCardMenu.shouldRender = ({
   );
 
   if (isPublicOrEmbedded) {
-    return downloadsEnabled;
+    return downloadsEnabled && !!result?.data && !result?.error;
```

Comment:
Also, it doesn't seem this line change has a test coverage since the function is used inside `DashCardVisualization` and that component doesn't seem to have a unit test. Also, if it were to be tested by E2E tests, some tests should have failed already.

https://github.com/metabase/metabase/blob/6045f2d22aea89e3689eed5251b731c7fa925f4b/frontend/src/metabase/dashboard/components/DashCard/DashCardVisualization.tsx#L203-L210

Can you add a test that tests this line change?

Reply:
@WiNloSt, great callouts, thank you! Please take another look, I've addressed your comments

Reply:
FWIW, we should keep the `!!result.data && !result.error` check. We can have a response like that:

```json
{
  "error": "Connection refused",
  "data": { "rows": [], "cols": [] }
}
```